### DISCUSSION TITLE
The size of exe file

### DISCUSSION THREAD

**Discussion by zy63600** [pyinstaller/pyinstaller#5718] [Help]
Is the packing process idempotent？If it is guaranteed that the size of the packaged result is the same？  looking forward to your answer very much ！

**Comment by bwoodsend** [comment]
As long as your environment doesn't change it should be bit for bit reproducible.

**Reply by bwoodsend** [reply]
If your environment does change then no.

**Reply by zy63600** [reply]
I have packaged my project many times by command, and the exe file size is different.

**Reply by zy63600** [reply]
the command :     pyinstaller.exe -F -w -i ./icon.ico -n detection ./main.py 

My business requires that after multiple packaging, the exe file size must be the same

is it possible?

**Reply by bwoodsend** [reply]
As long as the dependencies don't change, it'll stay the same size.

**Reply by zy63600** [reply]
I just packaged it 3 times in a row, the first time is 34880kb, the second time is 34886kb, the third time is 34887kb. 

the dependencies  what i used here

altgraph       0.17
future         0.18.2
pefile         2019.4.18
pip            20.1.1
PyInstaller    3.6
PyQt5          5.15.0
PyQt5-sip      12.8.0
pywin32-ctypes 0.2.0
setuptools     49.2.0
xlrd           1.2.0
xlwt           1.3.0

**Reply by zy63600** [reply]
Can you infer what affects the 10kb error?

**Reply by zy63600** [reply]
Do you need me to provide more packaging information?

**Reply by rokm** [reply]
> Can you infer what affects the 10kb error?

You can inspect executables with PyInstaller's `archive-viewer` utility (`pyi-archive_viewer.exe` that gets installed in Scripts directory in your environment) and see where the size difference is coming from (probably from `PYZ-00.pyz`).

But I see you're using PyInstaller 3.6, which is out-of-date, and has several issues w.r.t. reproducibility of builds. So try using `develop` branch instead (I think even 4.2 had some potential issues).

**Reply by zy63600** [reply]
thank you! I will have a try.

**Reply by zou-xingshun** [reply]
I tried to find the different file pkg-00. PKG in the build folder, but I couldn't parse it, so I updated PyInstaller to version4.2. In the version4.2, the size of each package seemed to be the same, I'm not sure why. However, when I use the DOS command 'fc /B 1.exe 2.exe' (1. Exe and 2. Exe are packaged with pure source code twice, only the time of packaging the command is inconsistent), it is different in almost all the memory addresses, I am not sure why.Based on my original inference, 1. Exe and 2. Exe should be exactly the same

**Reply by rokm** [reply]
Did you set `PYTHONHASHSEED`, as suggested by [docs](https://pyinstaller.readthedocs.io/en/stable/advanced-topics.html#creating-a-reproducible-build)?

**Reply by zou-xingshun** [reply]
I set PYTHONHASHSEED=0
